src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -806,6 +817,23 @@ public static ServerInfo GetServerVersion(IDbConnection connection)
                     serverInfo.OsVersion = reader[0].ToString();
                 });
 
+                int? isBigDataCluster = serverInfo.Options[ServerInfo.OptionIsBigDataCluster] as int?;
```

Comment:
What happens in C# if the key doesn't exist, can you remind me? I had thought it might throw.

Reply:
Dictionary will throw KeyNotFoundException: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.item?view=netframework-4.7.2